pkg/reconciler/v1alpha1/taskrun/resources/taskrunresolution.go

```diff
@@ -70,3 +72,23 @@ func ResolveTaskRun(tr *v1alpha1.TaskRunSpec, getTask GetTask, gr GetResource) (
 	}
 	return &rtr, nil
 }
+
+func getTaskSpec(taskRun *v1alpha1.TaskRun, getTask GetTask) (*v1alpha1.TaskSpec, string, error) {
+	taskSpec := &v1alpha1.TaskSpec{}
+	taskName := ""
+	if taskRun.Spec.TaskRef != nil && taskRun.Spec.TaskRef.Name != "" {
+		// Get related task for taskrun
+		t, err := getTask(taskRun.Spec.TaskRef.Name)
+		if err != nil {
+			return nil, taskName, fmt.Errorf("error when listing tasks for taskRun %s %v", taskRun.Name, err)
+		}
+		taskSpec = &t.Spec
+		taskName = t.Name
+	} else if taskRun.Spec.TaskSpec != nil {
+		taskSpec = taskRun.Spec.TaskSpec
+		taskName = taskRun.Name
+	} else {
+		return taskSpec, taskName, fmt.Errorf("TaskRun %s not providing TaskRef or TaskSpec", taskRun.Name)
```

Comment:
this happens to be the only line not covered by the unit tests and i actually wonder if we need it at all - at this point, we do happen to know that the taskSpec has been validated, and there should be no case like this

however maybe its safer to keep it - im fine with not covering the line in unit tests :D